src/sched/lthread.c

```diff
@@ -240,6 +240,17 @@ void lthread_run(void)
                     "[%4d] lthread_run(): lthread_resume (dequeue)\n",
                     lt ? lt->tid : -1);
                 _lthread_resume(lt);
+
+                // Break out of scheduler loop when previous thread triggered
+                // termination
+                if (_lthread_should_stop)
+                {
+                    SGXLKL_TRACE_THREAD(
+                        "[%4d] lthread_run(): quitting\n", lt ? lt->tid : -1);
+                    // We only need this ethread to leave the enclave
+                    _lthread_should_stop = false;
```

Comment:
I don't understand what's happening here.  One or more ethreads will reach this path, because this is a read then write, not a CAS.  Why do we want a nondeterministic number of ethreads to hit this path?

Reply:
Yes, this should be CAS. The idea is that we will only have a single ethread exit the enclave.